packages/components/src/handler.ts

```diff
@@ -175,13 +177,31 @@ export class CustomChainHandler extends BaseCallbackHandler {
     }
 
     handleLLMEnd() {
-        this.socketIO.to(this.socketIOClientId).emit('end')
+        /* send the end event from handleChainEnd */
+        // this.socketIO.to(this.socketIOClientId).emit('end')
```

Comment:
why is this commented out?

Reply:
Commented as the 'end' has to be sent on Chain End, Line 204. Sending on LLM End will mean that the end is NOT sent for cached responses, hence the need to send in Chain End.

Reply:
ah okay make sense